src/plugins/visualizations/server/migrations/visualization_saved_object_migrations.ts

```diff
@@ -1038,6 +1038,18 @@ const migrateTagCloud: SavedObjectMigrationFn<any, any> = (doc) => {
   return doc;
 };
 
+export const replaceIndexPatternReference: SavedObjectMigrationFn<any, any> = (doc) => ({
+  ...doc,
+  references: Array.isArray(doc.references)
+    ? doc.references.map((reference) => {
+        if (reference.type === 'index_pattern') {
+          reference.type = INDEX_PATTERN_SAVED_OBJECT_TYPE;
+        }
+        return reference;
+      })
+    : doc.references,
+});
+
```

Comment:
I saw this method above. Can we move this method in common folder and use in two places?

Reply:
let's leave it here, with the current design we do not have a common place for that.

Reply:
I agree with Alexey